balancer/grpclb/grpclb_test.go

```diff
@@ -1008,23 +1008,24 @@ func (s) TestGRPCLB_FallBackWithNoServerAddress(t *testing.T) {
 		grpc.WithTransportCredentials(&serverNameCheckCreds{}),
 		grpc.WithContextDialer(fakeNameDialer),
 	}
-	cc, err := grpc.Dial(r.Scheme()+":///"+beServerName, dopts...)
+	// Initialize the resolver state with the backend address.
+	r.InitialState(resolver.State{
+		Addresses:     []resolver.Address{{Addr: beLis.Addr().String()}},
+		ServiceConfig: r.CC.ParseServiceConfig(grpclbConfig),
+	})
+
+	cc, err := grpc.NewClient(r.Scheme()+":///"+beServerName, dopts...)
 	if err != nil {
-		t.Fatalf("Failed to dial to the backend %v", err)
+		t.Fatalf("Failed to create a newclient to the backend: %v", err)
 	}
 	defer cc.Close()
 	testC := testgrpc.NewTestServiceClient(cc)
 
 	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
 	defer cancel()
-	for i := 0; i < 2; i++ {
-		// Send an update with only backend address. grpclb should enter
-		// fallback and use the fallback backend.
-		r.UpdateState(resolver.State{
-			Addresses:     []resolver.Address{{Addr: beLis.Addr().String()}},
-			ServiceConfig: r.CC.ParseServiceConfig(grpclbConfig),
-		})
```

Comment:
We can't use `r.InitialState` here without changing the test logic. I suggest keep using using `cc.Connect` here. I've mentioned the 3 places that we can avoid `cc.Connect`, please only change those.